cluster/aws/templates/common.sh

```diff
@@ -15,11 +15,15 @@
 # limitations under the License.
 
 
-apt-get update
-apt-get install --yes curl
+until apt-get update; do
+  echo "== apt-get update failed, retrying =="
+  echo sleep 5
+done
+
+apt-get install -y curl
+apt-get dist-upgrade -y
```

Comment:
This is a little scary, because now we are running versions of code that we haven't necessarily tested.  But on the other hand, we are running whatever version of curl is the latest anyway (anything we apt-get), and we do want security updates.

I agree the right answer is probably to spin our own image (or otherwise carefully control exactly what software is installed) but I think this is better than not upgrading, particularly as we do install specific versions of docker & salt.